### ISSUE TITLE
terminal.integrated.profiles.windows not working in .vscode/settings.json take 2

### ISSUE THREAD

**Issue by Mair** [microsoft/vscode#123920]
- VS Code Version:  1.56.1
- OS Version: win 10 pro 64X

Workspace settings (.vscode/settings.json) terminal settings now give deprecated warnings

>This is deprecated, use `#terminal.integrated.defaultProfile.windows#` instead

however adding 

```json
  "terminal.integrated.profiles.windows": {
    "PowerShell -NoProfile": {
      "source": "PowerShell",
      "args": ["-NoProfile","echo",]
    }
  },
```
makes no difference and does not add any option in the terminal dropdown
Steps to Reproduce:

1. remove  "terminal.integrated.shell.windows": "cmd.exe",
2. add   "terminal.integrated.profiles.windows": {
    "PowerShell -NoProfile": {
      "source": "PowerShell",
      "args": ["-NoProfile","echo",]
    }
  },

this does work in the user settings but not in the workspace settings

@meganrogge  recommended the following

> You have to set terminal.integrated.allowWorkspaceConfiguration: true

however...
this unfortunately doesn't help my use case or make sense. Please let me explain
I have a template (https://github.com/Mair/create-esp32-app) that thousands of people are using, when the template is run it creates a vs-code project with the settings in place.
```
"terminal.integrated.shell.windows": "cmd.exe",
  "terminal.integrated.shellArgs.windows": ["/k", "C:\\esp\\idfv_4_2\\esp-idf\\export.bat"],
  "terminal.integrated.shell.linux": "/bin/bash",
  "terminal.integrated.shellArgs.linux": ["--init-file", "C:/esp/idfv_4_2/esp-idf/export.sh", "-i"],
  "terminal.integrated.shell.osx": "/bin/bash",
  "terminal.integrated.shellArgs.osx": ["--init-file", "C:/esp/idfv_4_2/esp-idf/export.sh", "-i"],
```
the paths to the export scripts are dynamically created when they run npx create-esp32-app
it is assumed that users will be able to run my app and create there project and it should just work without having to change other settings I cannot control.

In other words I have a template that thousands of users are using and this is a breaking change and I have no way of fixing this because I can't get a terminal to work correctly on someone else's vs code instance.

Does this issue occur when all extensions are disabled?: Yes/No
N/A

**Comment by meganrogge** [comment]
Can you do a trace log while creating a terminal? https://github.com/microsoft/vscode/wiki/Terminal-Issues#enabling-trace-logging

**Comment by meganrogge** [comment]
We have re-prioritized the `terminal.integrated.shell.windows` setting in VS code 1.56.2. You can upgrade to that version and they will work

**Comment by Mair** [comment]
[rendererLog.txt](https://github.com/microsoft/vscode/files/6488417/rendererLog.txt)

@meganrogge please find attached log file

> Regarding the profile not showing up in the dropdown, that's because you have to set "terminal.integrated.allowWorkspaceConfiguration":true. you should make sure you trust the origin of your folders though.

as mentioned. I cant tell thousands of users individually to updated there users settings. updating "terminal.integrated.allowWorkspaceConfiguration":true in workspace settings has no effect. 

I have updated to 15.5.2 and it is not resolved

@meganrogge I wish you wouldn't close this issue. I'm getting a lot of students reporting this error and it certainly is not resolved would you kindly re-open it
Thanks

**Comment by AlansCodeLog** [comment]
Yeah this doesn't work for me either-ish.

For debugging neither work.
For new terminals only `terminal.integrated.shell.windows` works.
For tasks: Both `terminal.integrated.shell.windows` and `terminal.integrated.defaultProfile.windows` work.

I'm on the latest Insiders build btw.

**Comment by Tyriar** [comment]
@Mair sorry about the hassle, as of today's Insiders build the `terminal.integrated.allowWorkspaceConfiguration` setting has been removed as the final solution to this is enabled with the new workspace trust feature which is a mechanism to say you trust an folder's origin that's shared across many components/extensions.

Additionally in insiders _a lot_ of improvements have happened around terminal profiles so it should be much more reliable now. The `shell`/`shellArgs` settings are still prioritized higher than profiles but will show a warning to migrate.

**Comment by Mair** [comment]
@Tyriar Thank you for that.
Any idea when it will be released?

**Comment by Tyriar** [comment]
We're currently testing it, so it should be released late next week.
